src/routes.js

```diff
@@ -41,21 +44,38 @@ router.post('/coffeegrams', async ctx => {
     shop: form['shop']
   });
 
-  ctx.request.flash('success', "Your new Coffeegram is up!");
+  ctx.request.flash('success', "Your new Coffeegram is up");
 
   ctx.redirect(`/users/${ctx.state.currentUser.username}`);
 });
 
 router.get('/coffeegrams/:id', async ctx => {
   var coffeegram = await Coffeegram.findById(ctx.params.id);
-  if(!coffeegram) {
+  if (!coffeegram) {
     throw new NotFound();
   }
-  await ctx.render('edit', {
+  await ctx.render('gram', {
     coffeegram
   });
 });
 
+router.post('/coffeegrams/:id', async ctx => {
+  var coffeegram = await Coffeegram.findById(ctx.params.id);
+  if (coffeegram.userId.toString() !== ctx.state.currentUser.id) {
```

Comment:
This will crash if someone went to a nonexistent coffeegram, perhaps you should do a `if (!coffeegram) { throw NotFound() }` n such?